tap-agent/src/agent/sender_accounts_manager.rs

```diff
@@ -106,19 +105,13 @@ impl Actor for SenderAccountsManager {
         }: Self::Arguments,
     ) -> std::result::Result<Self::State, ActorProcessingErr> {
         let (allocations_tx, allocations_rx) = watch::channel(HashSet::<Address>::new());
-        tokio::spawn(async move {
-            let mut indexer_allocations = indexer_allocations.clone();
-            while indexer_allocations.changed().await.is_ok() {
-                allocations_tx
-                    .send(
-                        indexer_allocations
-                            .borrow()
-                            .keys()
-                            .cloned()
-                            .collect::<HashSet<Address>>(),
-                    )
-                    .expect("Failed to update indexer_allocations_set channel");
-            }
+        watch_pipe(indexer_allocations.clone(), move |allocation_id| {
+            let allocations_tx = allocations_tx.clone();
```

Comment:
same here, you don't need to clone because you are not sending to the async block